diff --git a/mt_wifi/embedded/ap/ap_data.c b/mt_wifi/embedded/ap/ap_data.c
index f26a081..5210a08 100644
--- a/mt_wifi/embedded/ap/ap_data.c
+++ b/mt_wifi/embedded/ap/ap_data.c
@@ -4087,8 +4087,8 @@ INT ap_rx_pkt_foward(RTMP_ADAPTER *pAd, struct wifi_dev *wdev, PNDIS_PACKET pPac
 					but not to air if match following case:
 						a). pMbss->IsolateInterStaTraffic == TRUE
 				*/
-				to_air = TRUE;
-				to_os = FALSE;
+				to_os = TRUE;
+				to_air = FALSE;
 
 				if (pMbss->IsolateInterStaTraffic == 1)
 					to_air = FALSE;
